packages/firebase_vertexai/firebase_vertexai/example/lib/pages/token_count_page.dart

```diff
@@ -0,0 +1,106 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:flutter/material.dart';
+import 'package:firebase_vertexai/firebase_vertexai.dart';
+import '../widgets/message_widget.dart';
+
+class TokenCountPage extends StatefulWidget {
+  const TokenCountPage({super.key, required this.title, required this.model});
+
+  final String title;
+  final GenerativeModel model;
+
+  @override
+  State<TokenCountPage> createState() => _TokenCountPageState();
+}
+
+class _TokenCountPageState extends State<TokenCountPage> {
+  final List<MessageData> _messages = <MessageData>[];
+  bool _loading = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(8),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Expanded(
+              child: ListView.builder(
+                itemBuilder: (context, idx) {
+                  return MessageWidget(
+                    text: _messages[idx].text,
+                    isFromUser: _messages[idx].fromUser ?? false,
+                  );
+                },
+                itemCount: _messages.length,
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.symmetric(
+                vertical: 25,
+                horizontal: 15,
+              ),
+              child: Row(
+                children: [
+                  Expanded(
+                    child: ElevatedButton(
+                      onPressed: !_loading
+                          ? () async {
+                              await _testCountToken();
+                            }
+                          : null,
+                      child: const Text('Count Tokens'),
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> _testCountToken() async {
+    setState(() {
+      _loading = true;
+    });
+
+    const prompt = 'tell a short story';
+    var content = Content.text(prompt);
+    var tokenResponse = await widget.model.countTokens([content]);
+    final tokenResult = 'Count token: ${tokenResponse.totalTokens}, billable '
+        'characters: ${tokenResponse.totalBillableCharacters}';
+    _messages.add(MessageData(text: tokenResult, fromUser: false));
+
+    var contentResponse = await widget.model.generateContent([content]);
```

Comment:
These `var` look like they could be `final`

Reply:
done